### DISCUSSION TITLE
DOM recording in the web based forked vscode.

### DISCUSSION THREAD

**Discussion by aurk** [microsoft/vscode-discussions#440] [Extension Development QnA] [answered]
I've forked the VS Code. Now I'm trying to write extension(run only on browser). The primary purpose of the extension is when teacher click on the record button, It will start recording of the mouse movement, click, scroll. keyboard typing and when teacher click the stop recording then it will stop capturing event. In this regard, I need to access document object model(DOM). 

For student perspective, it will replay what teacher did. But I can handle it If I find the way how to access DOM.

**Comment by ghost** [comment, marked as answer]
This is only feasible in Monaco.
Access to VS Code DOM is Pandora's Box imho. It is a web-oriented "almost-IDE" rather than a webapp.
Will ask the extdevs what they think too :)

PS: As a suggestion you can now add extra data using [Screen Capture API](https://developer.mozilla.org/en-US/docs/Web/API/Screen_Capture_API/Using_Screen_Capture).

**Reply by olguzzar** [reply]
For more information, here's a link to the documentation specifying that [access to the DOM is restricted](https://code.visualstudio.com/api/extension-capabilities/overview#restrictions). I'm going to mark this thread as the answer.
